flocker/acceptance/endtoend/test_cli.py

```diff
@@ -0,0 +1,55 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Tests for the ``flocker-deploy`` command line tool.
+"""
+
+from twisted.trial.unittest import TestCase
+
+from ..testtools import require_flocker_cli, require_cluster
+from ...testtools import random_name
+
+
+class FlockerDeployTests(TestCase):
+    """
+    Tests for ``flocker-deploy``.
+    """
+    @require_flocker_cli
+    @require_cluster(1)
+    def test_deploy(self, cluster):
+        """
+        Deploying an application to one node and not another puts the
+        application where expected.
+        """
+        [node_1] = cluster.nodes
+        name = random_name(self)
+
+        minimal_deployment = {
+            u"version": 1,
+            u"nodes": {
+                node_1.reported_hostname: [name],
+            },
+        }
+
+        minimal_application = {
+            u"version": 1,
+            u"applications": {
+                name: {
+                    # Our config format doesn't support command-lines, so
+                    # we can't use standard image and need to pick
+                    # something small that will not immediately exit.
+                    u"image": u"openshift/busybox-http-app",
```

Comment:
The "Systems tests" design document says:

> End-to-end tests. These verify the components of the system work together correctly. As such we should minimize reliance on external software, since that introduces irrelevant complication

* So shouldn't this at least use a ClusterHQ controlled image? (like #1281)
* And ideally an image that doesn't have to be pulled from DockerHub. (Maybe call `docker build`  or `docker load`)
* And if neither of those options appeal, perhaps we should at least fix the image version so that upstream changes to the Dockerfile are less likely to break our tests.